### DISCUSSION TITLE
Remove same nearby text annotation

### DISCUSSION THREAD

**Discussion by rishkum** [pymupdf/PyMuPDF#1039] [Looking for help] [answered]
Hi Thanks for all your help. 

Is there a method on removing same nearby text annotation.  The annotation boxes have the same thing.
![image](https://user-images.githubusercontent.com/30025998/116719872-77522d80-a9d3-11eb-88bf-dd052c2d3b89.png). I am highlighting based on a topic. I just need to put the label of that topic next to the text once. I thought inst in text_instances was an entire sentence but it seems that it breaks that apart

My code is as follows:

```

import fitz
doc = fitz.open("output2.pdf")
inst_counter = 0
for i in range(len(sentec_lists)):
    print(f"we are on the {i} loop")
    for j in (range(len(sentec_lists[i]))):
        print(f"we are on the {j} inner loop")
        
        for pi in range(doc.pageCount):
            page = doc[pi]

            text = str(sentec_lists[i][j])
            text_instances = page.searchFor(text)
            
            #print( fills[i])
            for inst in text_instances:
                inst_counter += 1
                highlight = page.addHighlightAnnot(inst)
                #r = fitz.Rect(72, 72, 220, 100)
                loc = inst.tl
                stroke=(0,1,0)
                highlight.setColors({"stroke":fills[i]})
                highlight.update()
                pos = inst.tl + displ.tl
                annot = page.addTextAnnot(pos, list_of_words[i])

doc.save("output10.pdf", garbage=4, deflate=True, clean=True)
```

I tried putting an if condition in the inst in text_instances for loop but that just puts the TextAnnot once. Any ideas would be very helpful!
 
```
                if inst_counter == 1:
                    pos = inst.tl + displ.tl
                    highlight = page.addTextAnnot(pos, list_of_words[i])
```

**Comment by JorjMcKie** [comment, marked as answer]
You certainly can remove annotations - via `Page.delete_annot(annot)`.
So you only have to identify the right annotation for this, that's it.
Do this by checking / comparing the annotation rectangle or other properties like the text content.
Loop through a page's annotations via tha iterator `page.annots()`. Here you can also subselect the desired annotation type.
